Modules/ProtobufMessages/Sources/ServiceMessageHandlerAdapter.swift

```diff
@@ -16,45 +16,49 @@ public protocol ServiceEventsHandlerProtocol: AnyObject {
 /// - Subscribes as event handler to library events stream.
 /// - Transfer events from library to a value.
 ///
-public class ServiceMessageHandlerAdapter: NSObject {
+public class ServiceMessageHandlerAdapter {
     
-    public typealias Adapter = ServiceEventsHandlerProtocol
-    private(set) weak var value: Adapter?
+    private var handlers: [WeakHandler] = []
+    private var listener: ServiceMessageHandler?
     
-    public init(value: Adapter) {
-        self.value = value
-        super.init()
-        self.listen()
+    public static let shared = ServiceMessageHandlerAdapter()
+    
+    private init() {
+        listen()
     }
     
-    public override init() {
-        super.init()
+    public func addHandler(handler: ServiceEventsHandlerProtocol) {
+        handlers.append(WeakHandler(handler))
```

Comment:
it's a not big deal, but I think we should clear handlers with empty value sometimes (right here for example)

Reply:
agree